code/game/mecha/mecha_actions.dm

```diff
@@ -241,3 +244,30 @@
 	chassis.occupant_message("<span class='notice'>You switch to [equipment.name].</span>")
 	chassis.visible_message("[chassis] raises [equipment.name]")
 	send_byjax(chassis.occupant, "exosuit.browser", "eq_list", chassis.get_equipment_list())
+
+
+/datum/action/innate/mecha/mech_strafing_mode
+	name = "Toggle strafing"
+	button_overlay_icon_state = "strafe"
+
+/datum/action/innate/mecha/mech_strafing_mode/Grant(mob/living/L, obj/mecha/M)
+	. = ..()
+	button_overlay_icon_state = "[initial(button_overlay_icon_state)]_[M.strafing ? "on" : "off"]"
+	UpdateButtons()
+
+/datum/action/innate/mecha/mech_strafing_mode/Activate(force = FALSE)
+	if(force && chassis)
+		chassis.strafing = !chassis.strafing
+		button_overlay_icon_state = "[initial(button_overlay_icon_state)]_[chassis.strafing ? "on" : "off"]"
```

Comment:
I don't have the code on me rn, but I think you can include this in `UpdateButton()` for the action, or whatever proc gets called within `UpdateButtons()` so you don't manually need to set the icon state each time,